roofit/batchcompute/src/RooBatchCompute.cu

```diff
@@ -299,6 +300,11 @@ ReduceNLLOutput RooBatchComputeClass::reduceNLL(RooBatchCompute::Config const &c
    cudaStream_t stream = *cfg.cudaStream();
    constexpr int shMemSize = 2 * blockSize * sizeof(double);
 
+   for (auto span : {probas, weights, offsetProbas}) {
```

Comment:
This is nice to have in the code actually! Does this need to be in a `NDEBUG` macro? Otherwise, it's an empty loop because there is only an `assert` in it, and we'll get unused variables warnings.

Reply:
Good point, let's put in NDEBUG!

Reply:
Actually, I wrote a small test on [godbolt](https://godbolt.org)

The loop disappears, so OK to leave it like this @guitargeek ?

Reply:
That's ok too! Although I think there still be warnings about the unused variable